Add Design Assembly page

sowasred2012 opened the pull request:
## Done
- Added /design-assembly page
  - Added SSO
- Added link to /design-assembly on the homepage

## QA
- Visit https://webteam-canonical-com-93.demos.haus/ in a private browser
- Click the link on the "Design Assembly" card item
- You should be prompted to sign in with SSO
- When signed in, you should be redirected to /design-assembly
- The page should:
  - match the [design](https://www.figma.com/file/VSSGYamoulKk7sOwNfu21m/Hackaton%3A-Design-Assembly?node-id=0%3A1)
  - have working pagination
  - include links to colleague's directory profile when they're listed under "Moderator" or "Topic"

petesfrench reviewed (approved):
LGTM! Might be worth getting someone with more python expertise to take a look.

sowasred2012 commented:
QA -1 while SSO doesn't work on the demo

sowasred2012 commented:
@nottrobin would you mind giving this a once over?

nottrobin commented on webapp/design_assembly.py:
```diff
@@ -0,0 +1,66 @@
+
+# from webapp.sso import login_required
+
+DEPLOYMENT_ID = os.getenv(
```
Do we want to control this through environment variables or through code? I'm not sure it should be both, I think that makes it confusing to work out the state of any specific environment. Personally I think the team has much more visibility over code than we do over the Kubernetes secrets, so I'd probably suggest we just hard-code this and if we need to change it we do so with a pull request. Thoughts?

sowasred2012 replied:
This is down to my not being too comfortable with Python, we based a lot of this off the way /canonicool is [set up](https://github.com/canonical-web-and-design/webteam.canonical.com/blob/main/webapp/canonicool.py), it helped us get to the meat of things faster during the hack day - but I see what you mean, so I've updated this to be hardcoded.

nottrobin reviewed (approved):
This is really great work overall.

I have a couple of UX comments, but since this is just an internal page, I don't think we need to polish it too much before releasing, so I leave it up to the owner of this PR to decide whether to:

a) have the necessary discussions to decide what to do about these now, before merging; or
b) simply file them as issues and merge this PR now to get it live

I'm going to approve this anyway.

### Pagination behaviour

When you go to page 2, the information about "next session" disappears. This is surprising. I personally think that pagination should never fundamentally change a page. But I understand why you hide that section - because it pushes down the actual list content below the fold, so when you arrive  back at the top of page 2, you wouldn't even be able to see the results if you kept that section.

Which kind of highlights that that header section takes up *a lot* of space.

![image](https://user-images.githubusercontent.com/519935/162410371-bbabac2b-bfce-4757-9638-fb566e6b3d53.png)

Is there no way to merge the "next session" information into the top line of the header, and then leave all that information in-place on pagination?

### Empty content

The latest episode is missing both video and description.

![image](https://user-images.githubusercontent.com/519935/162410630-befd07af-1a76-4614-a1cd-488870aacb6c.png)

Is there any way to either:

1. Collapse the row so it takes up much less space; or
2. Add placeholder content to fill up the whitespace?

sowasred2012 commented:
@nottrobin thanks! The point about the video, yep, totally right, and I've updated it - it previously assumed the video was there, and held space open for it, so I've added a check to avoid that and reduce the height of the row in those cases.

The other item - I'm going to open an issue for it, I see your point, but I'd like to throw it out to other members of the team, especially those who were involved in the origin design during the hack day.
